### PR TITLE
Upgrade Grafana helm chart to 7.3.7 and application to 10.4.0

### LINKED ISSUE #2048: Upgrade Grafana to address cve-2023-6152
### Chart to upgrade

grafana/grafana

### Reason for upgrade

- [X] Security patching
- [ ] Upgrade to allow new features
- [ ] Upgrade to prevent future issues
- [ ] Other - _Write the reason below_

### Additional context

Upgrade the chart to a version that at least defaults to application version 10.3.3 or any later version which is not vulnerable to cve-2023-6152, see [here](https://grafana.com/blog/2024/02/14/grafana-security-release-medium-severity-security-fix-for-cve-2023-6152/)

### Acceptance criteria

- I checked the migration of the new chart version:
  - [ ] The upgrade does not require migration steps
  - [ ] The upgrade requires migration steps
- I tested the functionality of the new chart version:
  - [ ] The upgrade does not change current functionality
  - [ ] The upgrade changes current functionality

### PR DESCRIPTION
**Pull request by crssnd**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [x] kind/security      
- [ ] [kind/adr]()       

### Security notice
This upgrade mitigates the following medium security vulnerabilities:
- [CVE-2023-6152](https://github.com/advisories/GHSA-3hv4-r2fm-h27f)
- [CVE-2024-1442](https://github.com/advisories/GHSA-5mxf-42f5-j782)

### What does this PR do / why do we need this PR?

I moved client secrets and password into a Kubernetes secret, to avoid exposing them when rendering a chart.

- Fixes #2048 

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [x] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [x] The logs do not show any errors after the change
- Network Policy checks:
  - [x] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [x] Any changed pod is covered by Pod Security Admission
  - [x] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [x] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: helmfile.d/upstream/grafana/grafana/README.md [hunk 1]
```diff
@@ -223,7 +224,7 @@ need to instead set `global.imageRegistry`.
 | `admin.existingSecret`                    | The name of an existing secret containing the admin credentials (can be templated). | `""`                                 |
 | `admin.userKey`                           | The key in the existing admin secret containing the username. | `"admin-user"`                          |
 | `admin.passwordKey`                       | The key in the existing admin secret containing the password. | `"admin-password"`                      |
-| `serviceAccount.autoMount`                | Automount the service account token in the pod| `true`                                                  |
+| `serviceAccount.automountServiceAccountToken` | Automount the service account token on all pods where is service account is used | `false` |
```
Comment (Zash) [inline helmfile.d/upstream/grafana/grafana/README.md:227]: Does this change in default affect us in any way?

**Comment by Zash** [comment]
> Do correct me if I'm wrong. According to the link that I provided, the cve should have been mitigated in version 10.3.3 which seems to be the default app version in the [helm chart version 7.3.6](https://github.com/grafana/helm-charts/blob/grafana-7.3.6/charts/grafana/Chart.yaml#L4) :)

Correct. [Chart version 7.3.1](https://github.com/grafana/helm-charts/releases/tag/grafana-7.3.1) and later has the fix.

File: helmfile.d/upstream/grafana/grafana/README.md (hunk 1 above)
Comment (crssnd) [reply helmfile.d/upstream/grafana/grafana/README.md:227]: not from what I was able to test

**Comment by crssnd** [comment]
> > Do correct me if I'm wrong. According to the link that I provided, the cve should have been mitigated in version 10.3.3 which seems to be the default app version in the [helm chart version 7.3.6](https://github.com/grafana/helm-charts/blob/grafana-7.3.6/charts/grafana/Chart.yaml#L4) :)
> 
> I see that they in fixed [CVE-2024-1442](https://github.com/advisories/GHSA-5mxf-42f5-j782) in later versions. Is there anything preventing us from upgrading to 10.4, which is default in [7.3.7](https://github.com/grafana/helm-charts/blob/grafana-7.3.7/charts/grafana/Chart.yaml#L4)?

Not really, I thought that version is to new, and I preferred to upgrade to the latest available patch (as it fixes some other CVEs). I will check what changes were made in 10.4.0

File: helmfile.d/upstream/grafana/grafana/README.md (hunk 1 above)
Comment (OlleLarsson) [reply helmfile.d/upstream/grafana/grafana/README.md:227]: I'm guessing the datasource/dashboard reloader uses the token to query the k8s api for changes. This still works as intended?

Comment (OlleLarsson) [reply helmfile.d/upstream/grafana/grafana/README.md:227]: Seems to have been a lot of fuss about the change at least :smile: 
* https://github.com/grafana/helm-charts/pull/2977

Comment (crssnd) [reply helmfile.d/upstream/grafana/grafana/README.md:227]: The changes mentioned in this PR are already implemented in the chart version 7.3.7
I did make some changes to a dashboard and added a new datasource. No errors in the logs and the changes were visible in the UI as expected.

**Review by Zash — APPROVED** [review]

**Comment by anders-elastisys** [comment]
Since this fixes a CVE, should it be added as a security notice?

**Review by OlleLarsson — APPROVED** [review]
LGTM with Anders comment that we should add a mention that the two mentioned CVEs are resolved by this change :)

**Review by robinelastisys — APPROVED** [review]

**Pull request merged**